src/pages/LogInWithShortLivedTokenPage.js

```diff
@@ -82,15 +72,19 @@ class LogInWithShortLivedTokenPage extends Component {
         // exitTo is URI encoded because it could contain a variable number of slashes (i.e. "workspace/new" vs "workspace/<ID>/card")
         const exitTo = decodeURIComponent(lodashGet(this.props.route.params, 'exitTo', ''));
 
-        // User is signing in to a different account sign them out and sign in again
-        if (email !== this.props.session.email) {
+        const signedIn = this.props.session && this.props.session.authToken;
+        if (!signedIn) {
```

Comment:
Couple things... 

1. Not sure I understand why we need to handle the case of a user who is not signed in here (looks like that is covered in `componentDidMount()`). Is it possible for the user to be "not signed in" at this point?
2. Looks like we are back to using `signedIn` instead of `isUserSignedIn` or something equivalent.

Reply:
Great points, I should have caught those things. The are times that componentDidUpdate runs when the user is signed out, and we do not want to create a duplicate call to create a login. So how about this:

```
componentDidUpdate() {
     const isUserSignedIn = this.props.session && this.props.session.authToken;
      if (!isUserSignedIn) {
          return;
      }

      const email = lodashGet(this.props.route.params, 'email', '');

      // exitTo is URI encoded because it could contain a variable number of slashes (i.e. "workspace/new" vs "workspace/<ID>/card")
      const exitTo = decodeURIComponent(lodashGet(this.props.route.params, 'exitTo', ''));
     ...
}
```

Reply:
Yep, looks like we're already doing this to prevent it

https://github.com/Expensify/App/blob/bcc7630a3fa40ca2a94ec674ccaf75862a697d8d/src/pages/LogInWithShortLivedTokenPage.js#L76-L78

So changing it to this makes sense

```js
if (!this.props.session.authToken) {
    return;
}
```